@app/client/src/layout/SharedLayout.tsx

```diff
@@ -86,6 +98,14 @@ function SharedLayout({ title, noPad = false, children }: SharedLayoutProps) {
       ) : (
         children
       );
+    if (data && data.currentUser && forbidWhen == AuthRestrict.LOGGED_IN) {
+      return (
+        <StandardWidth>
+          <Redirect href={"/"} />
+        </StandardWidth>
+      );
+    }
```

Comment:
Nice! This needs to come after `useSharedLayoutQuery` of course, otherwise there's no `data` variable and branching is forbidden. What do you think about redirecting to the parent path rather than `/`? e.g. if they're on `/o/myorg/settings` we could just redirect to `/o/myorg` rather than `/`?

We also need to add something like:

```suggestion
    }
    if (data && data.currentUser === null && !loading && !error && forbidWhen === AuthRestrict.LOGGED_OUT) {
      return <Redirect href={`/login?next=${encodeURIComponent(fullHref)}`} />
    }
```

See https://github.com/graphile/starter/blob/82f5f8d5155877b3e22c65a4759f48ef5e326856/%40app/client/src/layout/SettingsLayout.tsx#L71 for an example of constructing this URL. Also, probably remove this functionality from SettingsLayout and move it wholesale to SharedLayout.

Reply:
yes, cool, I almost forgot with the possibility of that branch organizations-1 URL pattern

Reply:
Yeah sorry, really need to get that merged!

Reply:
Hi @benjie, I don't think it will easy to implement right now, because this `organization.slug` does not yet exist yet in the current branch, right? :)

Reply:
It shouldn't be specific to organizations; it just redirects to `?next={THE_CURRENT_URL_HERE}` - it should work for all pages existing and future.